test/utils/structs/EnumerableMap.test.js

```diff
@@ -14,132 +15,86 @@ const getMethods = (mock, fnSigs) => {
+        keyType,
+        keys: values[keyType],
+        values: values[valueType],
+
+        methods: getMethods(
+          mock,
+          keyType == 'address'
+            ? {
+                set: `$set(uint256,${keyType},${valueType})`,
+                get: `$get(uint256,${keyType})`,
+                tryGet: `$tryGet(uint256,${keyType})`,
+                remove: `$remove(uint256,${keyType})`,
+                length: `$length_EnumerableMap_${name}(uint256)`,
+                at: `$at_EnumerableMap_${name}(uint256,uint256)`,
+                contains: `$contains(uint256,${keyType})`,
+                keys: `$keys_EnumerableMap_${name}(uint256)`,
+              }
+            : {
+                set: `$set(uint256,${keyType},${valueType})`,
+                get: `$get_EnumerableMap_${name}(uint256,${keyType})`,
+                tryGet: `$tryGet_EnumerableMap_${name}(uint256,${keyType})`,
+                remove: `$remove_EnumerableMap_${name}(uint256,${keyType})`,
+                length: `$length_EnumerableMap_${name}(uint256)`,
+                at: `$at_EnumerableMap_${name}(uint256,uint256)`,
+                contains: `$contains_EnumerableMap_${name}(uint256,${keyType})`,
+                keys: `$keys_EnumerableMap_${name}(uint256)`,
+              },
+        ),
+
+        zeroValue: zeroValue[valueType],
+        events: {
+          setReturn: `return$set_EnumerableMap_${name}_${keyType}_${valueType}`,
+          removeReturn: `return$remove_EnumerableMap_${name}_${keyType}`,
+        },
+      },
+    ]),
+  );
 
-    beforeEach(async function () {
-      Object.assign(this, await loadFixture(fixture));
-    });
+  return { mock, env };
+}
 
-    shouldBehaveLikeMap(0n, 'uint256', {
-      setReturn: 'return$set_EnumerableMap_UintToUintMap_uint256_uint256',
-      removeReturn: 'return$remove_EnumerableMap_UintToUintMap_uint256',
-    });
+describe('EnumerableMap', function () {
+  beforeEach(async function () {
+    Object.assign(this, await loadFixture(fixture));
   });
 
-  // Bytes32ToUintMap
-  describe('Bytes32ToUintMap', function () {
-    const fixture = async () => {
-      const mock = await ethers.deployContract('$EnumerableMap');
-
-      const [keyA, keyB, keyC] = randomArray(generators.bytes32);
-      const [valueA, valueB, valueC] = randomArray(generators.uint256);
-
-      const methods = getMethods(mock, {
-        set: '$set(uint256,bytes32,uint256)',
-        get: '$get_EnumerableMap_Bytes32ToUintMap(uint256,bytes32)',
-        tryGet: '$tryGet_EnumerableMap_Bytes32ToUintMap(uint256,bytes32)',
-        remove: '$remove_EnumerableMap_Bytes32ToUintMap(uint256,bytes32)',
-        length: '$length_EnumerableMap_Bytes32ToUintMap(uint256)',
-        at: '$at_EnumerableMap_Bytes32ToUintMap(uint256,uint256)',
-        contains: '$contains_EnumerableMap_Bytes32ToUintMap(uint256,bytes32)',
-        keys: '$keys_EnumerableMap_Bytes32ToUintMap(uint256)',
+  // UintToAddressMap
+  for (const { name } of testTypes) {
```

Comment:
This will be goof for maintenance as the 'address -> uint' mapping was not being tested